src/plugins/opensearch_dashboards_react/public/table_list_view/table_list_view.tsx

```diff
@@ -498,25 +499,26 @@ class TableListView extends React.Component<TableListViewProps, TableListViewSta
   }
 
   renderListing() {
-    let createButton;
-    if (this.props.createItem) {
-      createButton = (
-        <EuiFlexItem grow={false}>
-          <EuiButton
-            onClick={this.props.createItem}
-            data-test-subj="newItemButton"
-            iconType="plusInCircle"
-            fill
-          >
-            <FormattedMessage
-              id="opensearch-dashboards-react.tableListView.listing.createNewItemButtonLabel"
-              defaultMessage="Create {entityName}"
-              values={{ entityName: this.props.entityName }}
-            />
-          </EuiButton>
-        </EuiFlexItem>
-      );
-    }
+    // Deprecated - createButton - this should be removed in 3.0 release
```

Comment:
I'm aligned that it's likely a better interface for consumers to provide their own button component rather than a create handler. But it seems relatively easy to support both - where we preferentially render the button component (this.props.createButton), if provided, and fallback to the default create button if `createItem` is provided. If neither is provided we shouldn't render anything. They're both already optional props, which seems right, so I don't see a huge need to deprecate in this case.

Reply:
Refactorred as recommended.